src/server/connections.ts

```diff
@@ -152,33 +153,59 @@ export default class Connections extends System {
      * Remove all connection data.
      */
     if (connection.isMain) {
-      this.channel(CHANNEL_DISCONNECT_PLAYER).delay(PLAYERS_REMOVE, connection.playerId);
-      secondConnectionId = this.storage.playerBackupConnectionList.get(connection.playerId);
 
-      this.storage.mainConnectionIdList.delete(connectionId);
-      this.storage.humanConnectionIdList.delete(connectionId);
-      this.storage.botConnectionIdList.delete(connectionId);
-      this.storage.playerMainConnectionList.delete(connection.playerId);
+      // Not confident this is a good approach.
+      // Delaying the disconnect event (if runnerDisconnectDelay is set) to keep players in the 
+      // game for a few seconds. This prevents runners from disconnecting / reloading to keep their leaderboard standing.
+
+      // Nuke the player's motion
+      if (this.config.connections.runnerDisconnectDelay > 0) {
+        let player = this.storage.playerList.get(connection.playerId)
+        player.keystate.DOWN = false
+        player.keystate.UP = false
+        player.keystate.LEFT = false
+        player.keystate.RIGHT = false
+        player.keystate.SPECIAL = false
+        player.keystate.FIRE = false
+        player.shield.current = false
+        player.delayed.BROADCAST_PLAYER_UPDATE = true
+      }
+
+      let ctx = this;
+      setTimeout(function() {
```

Comment:
Using the arrow function here (as elsewhere in the project) will avoid creating a closure.